src/NHibernate.Test/TransactionTest/TransactionFixture.cs

```diff
@@ -173,6 +173,35 @@ public void FlushFromTransactionAppliesToSharingSession()
 			}
 		}
 
+		[Test]
+		public void FlushFromTransactionAppliesToSharingStatelessSession()
```

Comment:
I have not added the equivalent test in system transaction, because the "flush from transaction completion" feature is basically broken with stateless sessions. A stateless session accumulates changes as a pending batched command, bound to the current connection. When flushing from a system transaction, this means the command has to execute from the NHibernate enlisted resource prepare phase, potentially after (or concurrently to) its connection own prepare phase.